src/ai/backend/storage/filebrowser/filebrowser.py

```diff
@@ -0,0 +1,208 @@
+from __future__ import annotations
+
+import logging
+from contextlib import asynccontextmanager
+from datetime import datetime
+from pathlib import Path
+from typing import Any, List, Mapping, Type
+from uuid import UUID
+
+import aiodocker
+import pkg_resources
+
+from ai.backend.common.logging import BraceStyleAdapter
+from ai.backend.common.validators import BinarySize
+from ai.backend.storage.abc import AbstractVolume
+from ai.backend.storage.context import Context
+from ai.backend.storage.utils import get_available_port
+from ai.backend.storage.vfs import BaseVolume
+
+from .config_browser_app import prepare_filebrowser_app_config
+from .database import FilebrowserTrackerDB
+
+BACKENDS: Mapping[str, Type[AbstractVolume]] = {
+    "vfs": BaseVolume,
+}
+
+
+log = BraceStyleAdapter(logging.getLogger(__name__))
+
+__all__ = (
+    "create_or_update",
+    "destroy_container",
+    "get_container_by_id",
+    "get_filebrowsers",
+    "get_network_stats",
+)
+
+
+@asynccontextmanager
+async def closing_async(thing):
+    try:
+        yield thing
+    finally:
+        await thing.close()
+
+
+async def create_or_update(
+    ctx: Context,
+    host: str,
+    vfolders: list[dict],
+) -> tuple[str, int, str]:
```

Comment:
Returning just tuple can make code readers confused about what the components do. Please replace it with `NamedTuple` or `TypedDict` so that readers can easily understand its role.